### ISSUE TITLE
Classes with all async/await methods are not correctly reported

### ISSUE THREAD

**Issue by tillig** [danielpalme/ReportGenerator#20]
I have found that if I have a class that _only_ has async/await methods in it, **that class always shows up as zero coverage in the ReportGenerator output**. I discovered this while working with Web API `DelegatingHandler` implementations, which basically only have async/await.

Example:

Here's a very simple `DelegatingHandler` class.

``` csharp

using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;

namespace ReportGeneratorRepro
{
  public class ReproHandler : DelegatingHandler
  {
    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
    {
      request.Properties.Add("demo", true);
      return await base.SendAsync(request, cancellationToken);
    }
  }
}
```

Note it **only** has async/await methods in it, so the compiler is going to generate that magic enumerable state machine to handle the syntactic sugar.

Here's an NUnit test that provides full coverage. (Well, full coverage except for a tiny bit on the generated async/await state machine.)

``` csharp

using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;
using NUnit.Framework;
using ReportGeneratorRepro;

namespace ReproUnitTests
{
  [TestFixture]
  public class ReproHandlerTests
  {
    [Test]
    public async void FullCoverage()
    {
      var request = new HttpRequestMessage();
      var expected = new HttpResponseMessage();
      var handler = new ReproHandler { InnerHandler = new EchoHandler(expected) };
      var invoker = new HttpMessageInvoker(handler, false);
      var actual = await invoker.SendAsync(request, new CancellationToken());
      Assert.AreSame(expected, actual);
      Assert.IsTrue((bool)request.Properties["demo"]);
    }

    private class EchoHandler : DelegatingHandler
    {
      private HttpResponseMessage _response;

      public EchoHandler(HttpResponseMessage response)
      {
        this._response = response;
      }

      protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
      {
        return Task.FromResult(this._response);
      }
    }
  }
}
```

**When run with coverage, I see 75% coverage for the handler and 85% coverage in the unit test.** (Not that I care a bunch about the coverage on the unit test proper, but it, too, only has async/await methods in it.) The nested `EchoHandler` class inside the test fixture shows 100% coverage.

![Coverage results inside Visual Studio](https://cloud.githubusercontent.com/assets/1156571/8111217/5b35dfa6-1015-11e5-92c1-3cf4fc883f8a.png)

Taking that .coverage report output and converting it to .coveragexml, I still see that **there's correct coverage reported**. It's not lost in the translation. For example, I see the generated SendAsync async/await state machine `MoveNext()` method getting reported with 75% line coverage:

    <function id="8272" token="0x6000004" name="MoveNext()" type_name="ReproHandler.&lt;SendAsync&gt;d__0" block_coverage="65.22" line_coverage="75.00" blocks_covered="15" blocks_not_covered="8" lines_covered="3" lines_partially_covered="1" lines_not_covered="0">

**All the numbers in the .coveragexml line up.**

However, if I run the ReportGenerator to get a report...

    ReportGenerator.exe "-reports:CoverageResults.coveragexml" "-targetdir:report"

...then **I actually see ZERO coverage for the whole thing**:

![HTML report showing zero coverage](https://cloud.githubusercontent.com/assets/1156571/8111301/dfa22e5c-1015-11e5-9b12-5958de68391e.png)

Not even the nested `EchoHandler` class in the unit test class is reported, even though it has full coverage and no async/await - the parent class, the unit test fixture, only has async/await so it seems to just ignore everything inside that class including nested classes.

**Comment by danielpalme** [comment]
Could you please provide the full ```CoverageResults.coveragexml```?

**Comment by tillig** [comment]
[Gist with CoverageResults.coveragexml is here.](https://gist.github.com/tillig/7e912f05e6214d2f7947)

**Comment by danielpalme** [comment]
Could you please test again with this beta:
https://www.nuget.org/packages/ReportGenerator/2.1.7-beta2

**Comment by tillig** [comment]
That looks pretty good to me! Problem solved with 2.1.7-beta2.
